### PR TITLE
Refactor: Remove global variable from module_pw

### PR DESCRIPTION
**Pull request by denghuilu**

### CODE REVIEW

**Review by dyzheng — APPROVED** [review]

File: source/module_pw/pw_basis.h [hunk 1]
```diff
@@ -260,6 +260,13 @@ class PW_Basis
     using resmem_int_op = psi::memory::resize_memory_op<int, psi::DEVICE_GPU>;
     using delmem_int_op = psi::memory::delete_memory_op<int, psi::DEVICE_GPU>;
     using syncmem_int_h2d_op = psi::memory::synchronize_memory_op<int, psi::DEVICE_GPU, psi::DEVICE_CPU>;
+
+    void set_device(std::string device_);
+    void set_precision(std::string precision_);
+
+protected:
+    std::string device = "cpu";
+    std::string precision = "double";
```
Comment (Qianruipku) [inline source/module_pw/pw_basis.h:270]: Please remove globalV::precision_flag = "single" in pw_test.cpp. 
Change `std::string precision = "double"` to

#ifdef __ENABLE_FLOAT_FFTW
    std::string precision = "single"
#else
    std::string precision = "double"
#endif

Then the segment fault can be solved.

**Review by Qianruipku — CHANGES_REQUESTED** [review]

File: source/module_pw/pw_basis_k_big.h [hunk 2]
```diff
@@ -22,6 +22,13 @@ class PW_Basis_K_Big: public PW_Basis_K
         by = 1;
         bz = 1;
     }
+    PW_Basis_K_Big(std::string device_, std::string precision_)
```
Comment (Qianruipku) [inline source/module_pw/pw_basis_k_big.h]: PW_Basis_K_Big(std::string device_, std::string precision_):PW_Basis(device_, precision_){}

Comment (Qianruipku) [reply source/module_pw/pw_basis_k_big.h]: Please also add 
PW_Basis_K(std::string device_, std::string precision_):PW_Basis(device_, precision_){}
in pw_basis.cpp
and 
PW_Basis_K_Big(std::string device_, std::string precision_):PW_Basis(device_, precision_){}
in pw_basis_k_big.h

Comment (denghuilu) [reply source/module_pw/pw_basis_k_big.h]: Added.

File: source/module_pw/pw_basis.h (hunk 1 above)
Comment (denghuilu) [reply source/module_pw/pw_basis.h:270]: Fixed.

**Review by Qianruipku — APPROVED** [review]

**Pull request merged**